diff -Nru libnl-3.2.21.orig/lib/nl.c libnl-3.2.21/lib/nl.c
--- libnl-3.2.21.orig/lib/nl.c	2014-01-21 17:14:46.017836475 +0100
+++ libnl-3.2.21/lib/nl.c	2014-01-21 17:15:48.315460504 +0100
@@ -597,7 +597,7 @@
 		flags |= MSG_PEEK | MSG_TRUNC;
 
 	if (page_size == 0)
-		page_size = getpagesize();
+		page_size = getpagesize() * 4;
 
 	iov.iov_len = sk->s_bufsize ? : page_size;
 	iov.iov_base = malloc(iov.iov_len);
